component/http/cache/cache.go

```diff
@@ -0,0 +1,349 @@
+package cache
+
+import (
+	"fmt"
+	"hash/crc32"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/beatlabs/patron/cache"
+	"github.com/beatlabs/patron/log"
+)
+
+type validationContext int
+
+const (
+	// validation due to normal expiry in terms of ttl setting
+	ttlValidation validationContext = iota + 1
+	// maxAgeValidation represents a validation , happening due to max-age  Header requirements
+	maxAgeValidation
+	// minFreshValidation represents a validation , happening due to min-fresh  Header requirements
+	minFreshValidation
+
+	// HeaderCacheControl is the Header key for cache related values
+	// note : it is case-sensitive
+	HeaderCacheControl = "Cache-Control"
+	// HeaderETagHeader is the constant representing the Etag http header
+	HeaderETagHeader = "Etag"
+
+	cacheControlMinFresh     = "min-fresh"
+	cacheControlNoCache      = "no-cache"
+	cacheControlNoStore      = "no-store"
+	cacheControlOnlyIfCached = "only-if-cached"
+	cacheControlEmpty        = ""
+
+	headerCacheMaxAge    = "max-age"
+	headerMustRevalidate = "must-revalidate"
+	headerWarning        = "Warning"
+)
+
+var metrics cacheMetrics
+
+func init() {
+	metrics = newPrometheusMetrics()
+}
+
+// NowSeconds returns the current unix timestamp in seconds
+var NowSeconds = func() int64 {
+	return time.Now().Unix()
+}
+
+// validator is a conditional function on an objects age and the configured ttl
+type validator func(age, ttl int64) (bool, validationContext)
+
+// expiryCheck is the main validator that checks that the entry has not expired e.g. is stale
+var expiryCheck validator = func(age, ttl int64) (bool, validationContext) {
+	return age <= ttl, ttlValidation
+}
+
+// cacheControl is the model of the request parameters regarding the cache control
+type cacheControl struct {
```

Comment:
since we live already in cache package the name can be
```suggestion
type control struct {
```